Tests/MongoSwiftTests/MongoCursorTests.swift

```diff
@@ -23,7 +23,7 @@ final class MongoCursorTests: MongoSwiftTestCase {
             // insert and read out one document
             try coll.insertOne(doc1)
             cursor = try coll.find()
-            var results = Array(cursor)
+            var results = try Array(cursor.all())
```

Comment:
we are implicitly testing it now via the other methods we call which depend on it, but I think it would be good to add a test somewhere in this file that explicitly tests out the updated `next()` method and consumes the returned `Result`s.